src/main/java/address/controller/RootLayoutController.java

```diff
@@ -119,6 +121,8 @@ private void handleSaveAs() {
         }
 
         modelManager.setPrefsSaveLocation(file.getPath());
+        // TODO temp patch to create cloud file if non found
+        raise(new CreateAddressBookOnRemoteRequestEvent(new CompletableFuture<>(), file.getName()));
```

Comment:
I assume in the permanent version controllers will not deal with `CompletableFuture`s?